nodebuilder/tests/tastora/blob_test.go

```diff
@@ -31,319 +34,432 @@ func TestBlobTestSuite(t *testing.T) {
 
-	bridgeNode := s.framework.GetBridgeNodes()[0]
-	client := s.framework.GetNodeRPCClient(ctx, bridgeNode)
+		// Reconstruct the namespace for this submission
+		namespaceBytes := bytes.Repeat([]byte{byte(0x30 + submissionIdx)}, 10)
+		namespace, err := share.NewV0Namespace(namespaceBytes)
+		s.Require().NoError(err)
 
-	namespace, err := share.NewV0Namespace(bytes.Repeat([]byte{0x04}, 10))
-	s.Require().NoError(err)
+		for nodeType, client := range clients {
+			s.Run(fmt.Sprintf("ConcurrentRetrieval_%s_%s", nodeType, submissionID), func() {
+				allBlobs, err := client.Blob.GetAll(ctx, height, []share.Namespace{namespace})
+				s.Require().NoError(err, "%s should retrieve blob for %s", nodeType, submissionID)
+				s.Assert().NotEmpty(allBlobs, "%s should find blobs for %s", nodeType, submissionID)
+
+				if len(allBlobs) > 0 {
+					expectedData := []byte(fmt.Sprintf("Concurrent submission %d data", submissionIdx))
+					retrievedData := bytes.TrimRight(allBlobs[0].Data(), "\x00")
+					s.Assert().Equal(expectedData, retrievedData,
+						"%s should return correct data for %s", nodeType, submissionID)
+				}
+			})
+		}
+	}
 
-	// Try to get non-existent blob
-	nonExistentCommitment := bytes.Repeat([]byte{0xFF}, 32)
-	_, err = client.Blob.Get(ctx, 1, namespace, nonExistentCommitment)
-	s.Require().Error(err)
-	s.Assert().Contains(err.Error(), "blob: not found")
+	s.T().Logf("✅ All %d concurrent blob operations verified across all nodes", numSubmissions)
 }
 
-// TestBlobGetAll_ValidNamespace tests GetAll API with valid namespace
-func (s *BlobTestSuite) TestBlobGetAll_ValidNamespace() {
-	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second) // Increased timeout
+// TestBlobProofConsistency validates that blob inclusion proofs are consistent
+// across different nodes in the network
+func (s *BlobTestSuite) TestBlobProofConsistency() {
+	ctx, cancel := context.WithTimeout(context.Background(), 4*time.Minute)
 	defer cancel()
 
 	bridgeNode := s.framework.GetBridgeNodes()[0]
-	client := s.framework.GetNodeRPCClient(ctx, bridgeNode)
+	lightNode := s.framework.NewLightNode(ctx)
 
-	// Create test namespace
-	namespace, err := share.NewV0Namespace(bytes.Repeat([]byte{0x05}, 10))
-	s.Require().NoError(err)
+	bridgeClient := s.framework.GetNodeRPCClient(ctx, bridgeNode)
+	lightClient := s.framework.GetNodeRPCClient(ctx, lightNode)
 
-	nodeAddr, err := client.State.AccountAddress(ctx)
+	// Create and submit test blob
+	namespace, err := share.NewV0Namespace(bytes.Repeat([]byte{0x05}, 10))
 	s.Require().NoError(err)
 
-	// Submit multiple blobs to same namespace
-	data1 := []byte("GetAll test data 1")
-	data2 := []byte("GetAll test data 2")
+	blobData := []byte("Proof consistency test data")
+	nodeBlobs := s.createBlobsForSubmission(ctx, bridgeClient, namespace, blobData)
 
-	libBlob1, err := share.NewV1Blob(namespace, data1, nodeAddr.Bytes())
+	txConfig := state.NewTxConfig(state.WithGas(250_000), state.WithGasPrice(5000))
```

Comment:
let's not do config manipulation if we do not need it in the context of the test - keep test as simple as possible.